src/core/core.ts

```diff
@@ -0,0 +1,58 @@
+import defaults from './defaults';
+import {
+	Context,
+	Definition,
+	ZOD_INSTANCE_IDENTIFIER,
+	ZOD_TYPE_IDENTIFIER,
+} from './generator';
+import { Utils } from './utils';
+
+export interface Config {
+	seed?: number;
+	defaults?: Partial<typeof defaults>;
+}
+
+export class Core {
+	#generators: Definition<any>[] = [];
+
+	readonly seed: number = 0;
+	readonly defaults = defaults;
+	readonly utils = new Utils(this);
+
+	constructor(config?: Config) {
+		this.defaults = { ...this.defaults, ...config?.defaults };
+		if (config?.seed) this.seed = config.seed;
+	}
+
+	register(generators: Definition<any>[]) {
+		this.#generators = generators.concat(this.#generators);
+		return this;
+	}
+
+	generate(schema: Zod.ZodTypeAny, ctx: Context): unknown {
```

Comment:
We should try to make `context` (and `path`) optional for a consumer.

```suggestion
	generate(schema: Zod.ZodTypeAny, context?: Context): unknown {
```

Because this is easier

```ts
core.generate(PersonSchema)
```

Instead of having to do the this

```ts
core.generate(PersonSchema, {path: []})
```

Reply:
I agree, I want to simplify it. The only place this isn't optional is when a consumer is making their own Generator for a map/array/object/record/etc. I'm sure there's a better pattern for making this known, but I haven't come up with it yet 🙃

Reply:
I refactored so this is now optional. There is now a default context with an empty path `{ path: [] }` if you don't provide one to the `generate` method.

```ts
core.generate( schema ) // implicit context of { path: [] } 
```